### PR TITLE
Make AuthorizationEnforcementService a daemon thread. Name it as "aut…

### PR DESCRIPTION
**Pull request by bdmogal**
…horization-enforcement-service" so it can be identified. Add log statement after shutdown is complete.

Tested that workflow completes successfully with this change, and does not without it. Running integration tests on a cluster now. Shouldn't need a unit test build for this change, but ran relevant authorization unit tests locally to be sure.

### CODE REVIEW

File: cdap-security/src/main/java/co/cask/cdap/security/authorization/DefaultAuthorizationEnforcementService.java [hunk 1]
```diff
@@ -119,9 +124,20 @@ public void enforce(EntityId entity, Principal principal, Action action) throws
   }
 
   @Override
+  protected ScheduledExecutorService executor() {
+    executor = Executors.newSingleThreadScheduledExecutor(
```
Comment (gokulavasan) [inline cdap-security/src/main/java/co/cask/cdap/security/authorization/DefaultAuthorizationEnforcementService.java:128]: Why do we need a ScheduledExecutorService? Do we run any task periodically?

Comment (gokulavasan) [inline cdap-security/src/main/java/co/cask/cdap/security/authorization/DefaultAuthorizationEnforcementService.java:128]: Oh it extends AbstractScheduledService, probably for refreshing ACL cache?

Comment (bdmogal) [inline cdap-security/src/main/java/co/cask/cdap/security/authorization/DefaultAuthorizationEnforcementService.java:128]: yup

**Comment by sagarkapare** [comment]
LGTM 👍

**Comment by bdmogal** [comment]
Integration tests passed (apart from tests that were not supposed to run on the cluster where they ran). Merging.

**Pull request merged**